client/asset/btc/btc.go

```diff
@@ -5907,3 +6015,15 @@ func (btc *baseWallet) ConfirmRedemption(coinID dex.Bytes, redemption *asset.Red
 		CoinID: coin.ID(),
 	}, nil
 }
+
+// writeRecycledAddrsToFile should be called with the recycledAddrMtx locked.
+func (btc *baseWallet) writeRecycledAddrsToFile() {
+	addrs := make([]string, 0, len(btc.recycledAddrs))
+	for addr := range btc.recycledAddrs {
+		addrs = append(addrs, addr)
+	}
+	contents := []byte(strings.Join(addrs, "\n"))
+	if err := os.WriteFile(btc.recyclePath, contents, 0644); err != nil {
```

Comment:
Missed this in last review, sorry, but I think the file should be 0600, not 0644 since it contains information that could be considered a privacy leak.